### DISCUSSION TITLE
the-difference-between-getservice-and-getrquiredservice-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#6] [Announcements]
# the-difference-between-getservice-and-getrquiredservice-in-asp-net-core/

This post looks at the GetService() and GetRequiredService() methods of the ASP.NET Core DI container, the difference between them and which you should use

https://andrewlock.net/the-difference-between-getservice-and-getrquiredservice-in-asp-net-core/

**Comment by DBJDBJ** [comment]
In that case, I assume, a program without a single `GetService` or `GetRequiredService` is a good program. Excuse my ignorance, what might be then the "DI hello world", in that case? Simple console Main would do..

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 18 2018, 08:39, in reply to [Paw Baltzersen](https://disqus.com/by/pawbaltzersen/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/6#discussioncomment-6046163)</em>

---

<p>Great point, I do! :)</p>

[a comment by an automated account is not shown]

**Comment by DBJDBJ** [comment]
@eduarddejong @andrewlock 

"...related to non-ASP.NET..."; it is probably an assumption too big, one would base all communications on HTTP. "inter-component" is not public and might utilize faster robust messaging, for example.  Just a valid example for a discussion on architecting extremely small and fast endpoints where inside one does not need sometimes baroque DI + ASP.NET scaffolding.  Think inter-container Linux services, implemented on  .NET Core.
